### ISSUE TITLE
Custom XYTileSource not Working on v6.1.7-6.1.11

### ISSUE THREAD

**Issue by DimitrisMantas** [osmdroid/osmdroid#1771]
Hi,

I'm currently using an custom map tile source foung [here](http://randochartreuse.free.fr/mobac2.x/documentation/#bsh) along with MobAC to create an SQLite database for offline use.

However, whenever I point osmdroid to it, i.e.,

``` java
mapView.setTileSource(
    new XYTileSource("OpenStreetMap Standard Tiles", 0, 15,256, ".png",
        new String[]{""}, "© OpenStreetMap contributors")
);
```

I just get a blank grid with no relevant warnings or erros on Logcat or the "Run" tab in Android Studio.

This behavior is the same for v6.1.7-6.11, but the above code works on v6.1.6 and below.

I have tried switching to Maperitive and using a ZIP archive instead of a SQLite database, but the issue persists.

What am I doing wrong?

## Issue Type

[ x] Question
[ x] Bug
[ ] Improvement
[ ] Build system related
[ ] Performance
[ ] Documentation

### If it's a bug, version(s) of android this affects:

All

### Version of osmdroid the issue relates to:

6.1.7-6.1.11

**Comment by monsieurtanuki** [comment]
Hi @DimitrisMantas!
The problem is that you don't provide tile URL prefixes.
Please have a look at the code in [`TileSourceFactory.java`](https://github.com/osmdroid/osmdroid/blob/master/osmdroid-android/src/main/java/org/osmdroid/tileprovider/tilesource/TileSourceFactory.java), where strings like `https://a.tile.openstreetmap.org/` will be automatically appended with z, x and y, something like https://a.tile.openstreetmap.org/5/15/13.png
```java
public static final OnlineTileSourceBase MAPNIK = new XYTileSource("Mapnik",
        0, 19, 256, ".png", new String[]{
        "https://a.tile.openstreetmap.org/",
        "https://b.tile.openstreetmap.org/",
        "https://c.tile.openstreetmap.org/"}, "© OpenStreetMap contributors",
        new TileSourcePolicy(2,
                TileSourcePolicy.FLAG_NO_BULK
                        | TileSourcePolicy.FLAG_NO_PREVENTIVE
                        | TileSourcePolicy.FLAG_USER_AGENT_MEANINGFUL
                        | TileSourcePolicy.FLAG_USER_AGENT_NORMALIZED
        ));
// max concurrent thread number is 2 (cf. https://operations.osmfoundation.org/policies/tiles/)
```

**Comment by DimitrisMantas** [comment]
Hi @monsieurtanuki!

Maybe it's a bit of a silly question, but since my tiles are stored offline on `sdcard/osmdroid/`, what URL should I provide?

Thanks.

**Comment by monsieurtanuki** [comment]
Good point! My suggestion: there is a sample osmdroid app on Google Play Store, with many use cases. Download that app, check which demo suits your own case, and have a look at the corresponding code.
